subiquity/server/runner.py

```diff
@@ -23,23 +26,47 @@ class LoggedCommandRunner:
 
     def __init__(self, ident):
         self.ident = ident
+        self.env_whitelist = [
+            "PATH", "PYTHONPATH",
+            "PYTHON",
+            "TARGET_MOUNT_POINT",
+            "SNAP",
+        ]
 
-    async def start(self, cmd):
-        proc = await astart_command([
-            'systemd-cat', '--level-prefix=false', '--identifier='+self.ident,
-            ] + cmd)
-        proc.args = cmd
+    def _forge_systemd_cmd(self, cmd: List[str], private_mounts: bool) \
+            -> List[str]:
+        """ Return the supplied command prefixed with the systemd-run stuff.
+        """
+        prefix = [
+            "systemd-run",
+            "--wait", "--same-dir",
+            "--property", f"SyslogIdentifier={self.ident}",
```

Comment:
I think for ubuntu-drivers we're going to want the output of a command run like this, but I think we can do that by giving the unit created by systemd-run a unique name and then reading the journal afterwards.

Reply:
That's right. We can use something like:
```bash
unit_name="subiquity-run-$(uuidgen)"
systemd-run --unit "$unit_name" ...
```
 Another way to do it would be to read from `stderr` when running `systemd-run` since it prints out the unit name that it generates from us:
```
systemd-run ls
Running as unit: run-rdb7cd9bb2f824b2f8fd7df2625a14ea3.service
```
Also, the first approach I considered was to use `systemd-run --pipe` which would pipe `stdout` and `stderr`back to the caller.
```systemd-run --pipe ls /root
Running as unit: run-u683.service
bin   cdrom  etc   lib	  lib64   lost+found  mnt  proc  run   snap  sys  usr
boot  dev    home  lib32  libx32  media       opt  root  sbin  srv   tmp  var
```
However with `--pipe`, the output does not go into the journal. I still think it's a good option.

Any preference?

Reply:
I didn't know about --pipe, that's probably easier indeed.

Having everything end up in the journal would be useful in some ways but probably not worth the complication.